src/Movies/index.js

```diff
@@ -1,52 +1,65 @@
-import React, { useState, useEffect } from "react";
+import React, { useEffect } from "react";
 import axios from "axios";
 import MovieSelectOptions from "./MovieSelectOptions";
 import SelectedMovie from "./SelectedMovie";
 import Loader from "../Loader";
-import { sortFunctionAsc, baseUrl } from "../utils";
+import {
+  sortFunctionAsc,
+  baseUrl,
+  retrieveCacheValues,
+  cacheValues
+} from "../utils";
+import { useMovieState } from "./store";
+import { actions } from "./actions";
 
 const Movies = () => {
-  const [movies, setMovies] = useState(null);
-  const [errorText, setErrorText] = useState(null);
-  const [selectedMovie, setSelectedMovie] = useState(null);
-  const [loading, setLoading] = useState(false);
+  const { state, dispatch } = useMovieState();
+  const { movieLoading, movieError, movies, selectedMovie } = state;
 
   const handleChange = e => {
     e.preventDefault();
     const index = e.target.value;
-    setSelectedMovie(movies[index]);
+    dispatch({ type: actions.SELECTED_MOVIE, payload: movies[index] });
   };
 
   useEffect(() => {
     const fetchMovies = async () => {
+      const key = "movies";
+      const cachedMovies = retrieveCacheValues(key);
```

Comment:
If there has been a network error, movies will be set as [] and cachedMovies will have been truthy, therefore not trying it again. and the app will become unusable unless one empties the local storage manually.

Reply:
You can replicate this by throttling or disconnecting your connection in your devtools and will see that movies will be set as []